src/components/data_display/routes.ts

```diff
@@ -152,24 +156,80 @@ const routeDataViewFilterList = async (request: any, reply: any) => {
 
   const dataView = dataViews[0]
 
-  // Check column exists on highest priority dataView
-  if (!dataView.tableViewIncludeColumns?.includes(columnName))
-    throw new Error(`Column "${dataViewCode}" not available on data view: "${dataView.title}"`)
+  // TO-DO: Create search filters for types other than string (number, bool, array)
+  const searchFilter =
+    searchText === ''
+      ? {}
+      : {
+          or: searchFields.map((col: string) => ({
+            [col]: {
+              includesInsensitive: searchText,
+            },
+          })),
+        }
+
+  const gqlFilters = { ...getFilters(dataView, userId, orgId), ...searchFilter }
+
+  const filterList = new Set()
 
-  // GraphQL query column including row_restrictions
-  const gqlFilters = getFilters(dataView, userId, orgId)
+  let fetchedCount = 0
+  let offset = 0
+  let moreResultsAvailable = true
 
-  const items = (
-    await queryFilterList(
+  while (filterList.size < config.filterListMaxLength) {
+    const { fetchedRecords, totalCount, error } = await queryFilterList(
       camelCase(getValidTableName(dataView.tableName)),
-      columnName,
+      searchFields,
       gqlFilters,
+      config.filterListMaxLength,
+      offset,
       authHeaders
     )
-  ).map((item: { [key: string]: unknown }) => item[columnName])
 
-  // TO-DO: How to handle very large lists?
-  return reply.send([...new Set(items)])
+    if (error) return reply.send(error)
+
+    fetchedCount += fetchedRecords.length
+
+    fetchedRecords.forEach((record: { [key: string]: unknown }) => {
+      // For some reason, if a single field value is `null`, Postgraphile
+      // returns null instead of an object with a null value on one field, like
+      // the rest of the results
+      if (record === null) {
+        filterList.add(null)
+        return
+      }
+      const values = Object.values(record)
+      values.forEach((value) => {
+        if (delimiter && typeof value === 'string') {
+          const splitString = value.split(delimiter).map((e) => e.trim())
+          splitString.forEach((string) => filterList.add(string))
+        } else filterList.add(value)
+      })
+    })
+
+    if (fetchedCount >= totalCount) {
+      moreResultsAvailable = false
```

Comment:
`moreResultsAvailable` allows us to tell the front-end if we've retrieved the full list or not, and display info in the UI accordingly.